src/index.js

```diff
@@ -1,42 +1,94 @@
+const todoInput = document.querySelector('#new-todo');
+const addTodoBtn = document.querySelector('#add-todo');
+const clearCheckedBtn = document.querySelector('#clear-checked');
+const todosList = document.querySelector('#todos');
 
-list.sort((a, b) => a.index - b.index);
+let todos = getTodos();
 
-const listCont = document.getElementById('list');
-const populate = () => {
-  for (let i = 0; i < list.length; i += 1) {
-    const checkcompleted = list[i].completed ? 'checked' : '';
-    const li = document.createElement('li');
-    li.classList.add('list_item');
-    li.innerHTML = `
-      <div class="list_con">
-        <input type="checkbox" ${checkcompleted}>
-        <p>${list[i].description}</p>
-      </div>
-                  
-      <button class="listBtn"><i class="fa-solid fa-ellipsis-vertical"></i></button>
-    
-    `;
-    listCont.appendChild(li);
+function createTodoElement(todo) {
+  const todoElement = document.createElement('li');
+  todoElement.classList.add('list_item');
+  todoElement.innerHTML = `
+    <div class='list_con'>
+      <input class="input1" type="checkbox" ${todo.completed ? 'checked' : ''}>
+      <span class='p' contenteditable="true" class="${todo.completed ? 'completed' : ''}">${todo.task}</span>    
+    </div>
+    <div>
+      <button class="btns edit-todo"><i class="fa-solid fa-ellipsis-vertical"></i></button>
+      <button class="btns remove-todo" style="display: none;"><i class="fa-solid fa-trash-can"></i></button>
+      <button class="btns save-todo" style="display: none;">Save</button>
+    </div>   
+  `;
+  const todoTextElement = todoElement.querySelector('span');
+  const editTodoButton = todoElement.querySelector('.edit-todo');
+  const removeTodoButton = todoElement.querySelector('.remove-todo');
+  const saveTodoButton = todoElement.querySelector('.save-todo');
+  todoTextElement.addEventListener('keydown', (event) => {
+    if (event.key === 'Enter') {
+      event.preventDefault();
+      saveTodoButton.click();
+    }
+  });
+  saveTodoButton.addEventListener('click', () => {
+    todo.task = todoTextElement.textContent.trim();
+    saveTodos(todos);
+    saveTodoButton.style.display = 'none';
+  });
+  todoElement.querySelector('input').addEventListener('change', () => {
+    todo.completed = !todo.completed;
+    saveTodos(todos);
+    if (todo.completed) {
+      todoTextElement.classList.add('completed');
+    } else {
+      todoTextElement.classList.remove('completed');
+    }
+  });
+  editTodoButton.addEventListener('click', () => {
+    todoTextElement.contentEditable = true;
+    todoElement.style.background = 'lightyellow';
+    editTodoButton.style.display = 'none';
+    removeTodoButton.style.display = 'inline-block';
+  });
+  todoElement.querySelector('button.remove-todo').addEventListener('click', () => {
+    const index = todos.indexOf(todo);
+    todos.splice(index, 1);
+    saveTodos(todos);
+    todos.forEach((todo, index) => {
+      todo.index = index;
+    });
```

Comment:
- [ ] According to requirements, deleting a task should update all remaining items' indexes, so they represent the current list order and are unique. Currently, as you can see in the attached image below, the indexes don't get updated after deleting. Please implement this change 🙏 
![image](https://user-images.githubusercontent.com/46839548/222867354-29a618ff-fc12-4700-acd8-5b49d21590f1.png)